playbooks/ansible/roles/sit.cephfs/tasks/main.yml

```diff
@@ -90,10 +90,50 @@
       command: /root/cephadm shell -- ceph config generate-minimal-conf
       register: ceph_conf
 
+    - name: Enable and create smb cluster using mgr module
+      when: config.be.variant == 'mgr'
+      block:
+        - name: Enable smb mgr module
+          command: /root/cephadm shell -- ceph mgr module enable smb
+
+        - name: Populate ceph.smb.usersgroups(siteacct) resource as YAML
+          template:
+            src: ceph.smb.usersgroups.yml.j2
+            dest: /etc/ceph/ceph.smb.usersgroups.yml
+
+        - name: Create the ceph.smb.usersgroups(siteacct) resource
+          shell: >
+            /root/cephadm shell -- ceph smb apply -i - < /etc/ceph/ceph.smb.usersgroups.yml
```

Comment:
Are you asking why don't we directly execute the following command?
```
/root/cephadm shell -- ceph smb apply -i /etc/ceph/ceph.smb.usersgroups.yml
```
At least I couldn't get it to work as `cephadm` couldn't locate the input file.

If not I didn't understand the question very well. Or is it like you were wondering why we generate a file and provide it as input rather than directly feeding the stdin with declarative yaml contents itself? In the latter case the task may not look so good.

Reply:
There's a [stdin](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/command_module.html#parameter-stdin) option to the `command` (and `shell`) module. I was wondering why through the bother of writing the template to yaml on the node when you can just render the template to a string for the `stdin` option. Something like (untested):

```yaml
- name: Foo
  command: cephadm shell ceph smb apply -i -
  stdin: "{% lookup('template', 'ceph.smb.usersgroups.yml.j2') %}"
```

Reply:
Cool. I wasn't aware of [ansible.builtin.template lookup](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/template_lookup.html) and the `stdin` option seems perfect. Thanks for letting me know.